crates/typst-library/build.rs

```diff
@@ -0,0 +1,14 @@
+use std::process::Command;
+
+fn main() {
+    let hash = Command::new("git")
+        .args(["rev-parse", "HEAD"])
+        .output()
+        .ok()
+        .filter(|output| output.status.success())
+        .and_then(|output| {
+            String::from_utf8(output.stdout.strip_suffix(b"\n").unwrap().into()).ok()
```

Comment:
Do we really need the full commit hash? What's the motivation for changing it from just having the first 8 chars?

Reply:
Well for one it feels better to give the user the whole thing and let them truncate it themselves if they want. It also lowers the chance of a collision.
Anyway, if we do end up giving out a short commit, I'd rather use `git rev-parse --short=8` instead of truncating the buffer manually. (Do you know how established that feature is, i.e. can we count on everyone building typst to have a recent enough git version that includes it?)

Reply:
I don't know. :/

Reply:
I just did a bit of digging and I can confirm that git rev-parse has had the `--short` flag since at least v1.7.9 and it didn't even exist in v1.7.0, so it's definitely safe to use. I'll probably open another PR to switch it over

Reply:
Oops! No, I was slightly wrong. rev-parse has existed since at least 1.0 and --short was introduced in 1.2